File: Java/FizzBuzz.java

import java.util.ArrayList;
import java.util.List;

// https://leetcode.com/problems/fizz-buzz/


    public List<String> fizzBuzz(int n) {
        List <String> s = new ArrayList<>();

        for (int i = 1; i <= n ; i++) {
            if( i % 15 == 0){
                s.add("FizzBuzz");
            }else if( i % 3 == 0){
                s.add("Fizz");
            }else if(i % 5 == 0){
                s.add("Buzz");
            }else{
                s.add(Integer.toString(i));
            }
        }
        return s;
    }

File: C++/projects/fizzBuzz.cpp
#include <bits/stdc++.h>
using namespace std;

vector<string> fizzBuzz(int n) {
    vector<string>res;
    int fizz=0,buzz=0;
    for(int i=1;i<=n;i++){
        fizz++;buzz++;
        if(fizz==3 && buzz==5){
            res.push_back("FizzBuzz");
            fizz=0;
            buzz=0;
        }
        else if(fizz==3){
            res.push_back("Fizz");
            fizz=0;
        }
        else if(buzz==5){
            res.push_back("Buzz");
            buzz=0;
        }
        else{
            res.push_back(to_string(i));
        }
    }
    return res;
}
    
int main(){
    int n = 3;
    vector<string> ans = fizzBuzz(n);
    for(auto it:ans){
        cout<< it << " ";
    }
	return 0;
}
